go-controller/pkg/ovn/egressip.go

```diff
@@ -1195,24 +1226,34 @@ func (oc *Controller) deletePodEgressIPAssignments(name string, statusesToRemove
 	podStatus, exists := oc.eIPC.podAssignment[podKey]
 	if !exists {
 		return nil
+	} else if podStatus.egressIPName != name {
+		// standby egressIP no longer matches this pod, update cache
+		podStatus.standbyEgressIPNames.Delete(name)
+		return nil
 	}
 	podIPs, err := util.GetPodCIDRsWithFullMask(pod)
 	if err != nil {
 		return err
 	}
 	for _, statusToRemove := range statusesToRemove {
-		klog.V(5).Infof("Deleting pod egress IP status: %v for EgressIP: %s and pod: %s/%s", statusToRemove, name, pod.Name, pod.Namespace)
+		klog.V(2).Infof("Deleting pod egress IP status: %v for EgressIP: %s and pod: %s/%s", statusToRemove, name, pod.Name, pod.Namespace)
 		if err := oc.eIPC.deletePodEgressIPAssignment(name, statusToRemove, podIPs); err != nil {
 			return err
 		}
-		delete(podStatus.egressStatuses, statusToRemove)
 		if err := oc.eIPC.addExternalGWPodSNAT(pod.Namespace, pod.Name, statusToRemove); err != nil {
 			return err
 		}
+		delete(podStatus.egressStatuses, statusToRemove)
 	}
 	// Delete the key if there are no more status assignments to keep
 	// for the pod.
-	delete(oc.eIPC.podAssignment, podKey)
+	if len(podStatus.egressStatuses) == 0 {
+		// pod could be managed by more than one egressIP
+		// so remove the podKey from cache only if we are sure
+		// there are no more egressStatuses managing this pod
+		klog.V(5).Infof("Deleting pod key %s from assignment cache", podKey)
```

Comment:
NIT: do we need V(5) here? Assuming this is not something that will happen often, I'm wondering id it can be V(2) or not V() at all.

Reply:
this will happen often right? everytime a pod moves around, deleted/created/egressIP re-assigns...